tests/test_z3_visitor.py

```diff
@@ -149,6 +159,25 @@ def not_in_empty_tuple(x: int):
 expected_list = [arithmetic_expected, boolean_expected, container_expected]
 
 
+def test_z3_vars():
```

Comment:
This test should have a much more specific name and description

Reply:
Also, move this test to after all of the other tests (which belong together in a block; see changes in #1061.